pkg/controller/llmisvc/router.go

```diff
@@ -96,41 +126,79 @@ func (r *LLMInferenceServiceReconciler) expectedHTTPRoute(ctx context.Context, l
 	}
 
-	// TODO(webhook): validate if referenced gateway exists
 	if llmSvc.Spec.Router != nil && llmSvc.Spec.Router.Gateway != nil {
 		log.FromContext(ctx).Info("Reconciling Gateway", "gateway", llmSvc.Spec.Router.Gateway)
 
 		// If Gateway is not managed (has .refs), re-attach the expected route to the referenced gateways
 		if llmSvc.Spec.Router.Gateway.HasRefs() {
 			httpRoute.Spec.CommonRouteSpec.ParentRefs = make([]gatewayapi.ParentReference, 0, len(llmSvc.Spec.Router.Gateway.Refs))
 			for _, ref := range llmSvc.Spec.Router.Gateway.Refs {
-				httpRoute.Spec.CommonRouteSpec.ParentRefs = append(httpRoute.Spec.CommonRouteSpec.ParentRefs, gatewayapi.ParentReference{
-					// TODO(api): With this structure we are missing the ability to narrow a section of targeted gateway by the route we are creating
-					// missing SectionName and Port
-					Name:      ref.Name,
-					Namespace: &ref.Namespace,
-					Group:     ptr.To(gatewayapi.Group("gateway.networking.k8s.io")),
-					Kind:      ptr.To(gatewayapi.Kind("Gateway")),
-				})
+				httpRoute.Spec.CommonRouteSpec.ParentRefs = append(httpRoute.Spec.CommonRouteSpec.ParentRefs, toGatewayRef(ref))
 			}
 		}
 	}
 
 	return httpRoute
 }
 
-func (r *LLMInferenceServiceReconciler) reconcileHTTPRoute(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService, expected *gatewayapi.HTTPRoute) error {
+func (r *LLMInferenceServiceReconciler) reconcileHTTPRoute(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService, expected *gatewayapi.HTTPRoute) (*gatewayapi.HTTPRoute, error) {
 	curr := &gatewayapi.HTTPRoute{}
 
 	err := r.Client.Get(ctx, client.ObjectKeyFromObject(expected), curr)
 	if err != nil && !apierrors.IsNotFound(err) {
-		return fmt.Errorf("failed to get HTTPRoute %s/%s: %w", expected.GetNamespace(), expected.GetName(), err)
+		return nil, fmt.Errorf("failed to get HTTPRoute %s/%s: %w", expected.GetNamespace(), expected.GetName(), err)
 	}
 	if apierrors.IsNotFound(err) {
-		return Create(ctx, r, llmSvc, expected)
+		return expected, Create(ctx, r, llmSvc, expected)
+	}
+
+	return expected, Update(ctx, r, llmSvc, curr, expected, semanticHTTPRouteIsEqual)
+}
+
+func (r *LLMInferenceServiceReconciler) updateRoutingStatus(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService, routes ...*gatewayapi.HTTPRoute) error {
+	logger := log.FromContext(ctx)
+
+	var urls []*apis.URL
+	for _, route := range routes {
+		discoverURL, err := DiscoverURLs(ctx, r.Client, route)
+		if IgnoreExternalAddressNotFound(err) != nil {
+			return fmt.Errorf("failed to discover URL for route %s/%s: %w", route.GetNamespace(), route.GetName(), err)
+		}
+		if discoverURL != nil {
+			urls = append(urls, discoverURL...)
+		}
+	}
+
+	slices.SortStableFunc(urls, func(a, b *apis.URL) int {
+		return cmp.Compare(a.String(), b.String())
+	})
+
+	externalURLs := FilterExternalURLs(urls)
+	if len(externalURLs) == 0 {
+		logger.Info("no public URL discovered")
```

Comment:
In raw deployment at least (and also Serverless but I need to confirm), we populate the status.url with the external URL if available, and if not we populate it with the internal url. Whereas the status.address always points to the internal url. I wonder if we should follow the same pattern here as well.

Reply:
Our initial thinking was that this field is external, everything else is in the .addresses. Should keep it aligned @pierDipi? In API docs we say `// URL of the publicly exposed service.`

Reply:
`status.addresses` is not `status.address`, `status.address` is internal only

Reply:
for reference [this](https://github.com/opendatahub-io/kserve/blob/master/pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/kube_ingress_reconciler.go#L138-L158) is what we have in ODH today for rawdeployment. 

in upstream the [logic](https://github.com/kserve/kserve/blob/master/pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/ingress_reconciler.go#L104-L112) is slightly different but I think the end result IIRC should be the same

Reply:
Thanks for the reference @VedantMahabaleshwarkar - I think we can follow-up on that in separate PR, WDYT?

Reply:
Yeah that is fine with me